File: app/src/main/java/me/gabeg/sicksends/SsTradRepository.java
package me.gabeg.sicksends;

import android.app.Application;

/**
 * Repository for trad problems.
 */
public class SsTradRepository
	extends SsProblemRepository<SsTradDao, SsTrad>
{

	/**
	 */
	public SsTradRepository(Application app)
	{
		super(app);
	}

	/**
	 * @see SsProblemRepository#buildDao(SsProblemDatabase)
	 */
	@Override
	protected void buildDao(SsProblemDatabase db)
	{
		this.mProblemDao = db.tradDao();
	}

}

File: app/src/main/java/me/gabeg/sicksends/db/trad/SsTradRepository.kt
package me.gabeg.sicksends.db.trad

import me.gabeg.sicksends.db.generic.SsGenericProblemRepository
import javax.inject.Inject

/**
 * Repository for trad problems.
 */
class SsTradRepository @Inject constructor(dao : SsTradDao)
	: SsGenericProblemRepository<SsTradProblem>(dao)
